### PR TITLE
Add bpn-discovery helm

### PR DESCRIPTION
**Pull request by tunacicek**
## Description
feat: Add bpn-discovery helm chart
## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: charts/umbrella/Chart.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ sources:
   - https://github.com/eclipse-tractusx/e2e-testing
 
 type: application
-version: 0.3.0
+version: 0.3.1
```
Comment (evegufy) [inline charts/umbrella/Chart.yaml:31]: As discussed, I'd bump the version to minor for adding a new product.

**Review by evegufy — CHANGES_REQUESTED** [review]
As discussed, regarding the failing helm test, I recommend to make the readiness probes configurable in the bpndiscovery helm chart, maybe you can also get some other improvement in, like you've done for dt [here](https://github.com/eclipse-tractusx/sldt-digital-twin-registry/pull/281)

File: charts/umbrella/Chart.yaml (hunk 1 above)
Comment (tunacicek) [reply charts/umbrella/Chart.yaml]: done

**Review by evegufy — CHANGES_REQUESTED** [review]
Hi @tunacicek thanks for making available a chart version with configurable probes, so I was able to the get more logs and noticed that the discovery services aren't able to find their corresponding databases. This originates from [SPRING_DATASOURCE_URL](https://github.com/eclipse-tractusx/sldt-bpn-discovery/blob/26c372c86ae703df9bb43dcbd41332c841c3b76b/charts/bpndiscovery/templates/postgres-init.yaml#L38).
Example:
Locally I have chosen the release name `disc`
which results in SPRING_DATASOURCE_URL: `jdbc:postgresql://disc-postgresql:5432/bpndiscovery`
But the name of the database with the nameOverride (which is also necessary to set) is `disc-bpndiscovery-postgresql`.
Once I change SPRING_DATASOURCE_URL to `jdbc:postgresql://disc--bpndiscovery-postgresql:5432/bpndiscovery` the service started up without issues.
I recommend to change the way how you determine [SPRING_DATASOURCE_URL](https://github.com/eclipse-tractusx/sldt-bpn-discovery/blob/26c372c86ae703df9bb43dcbd41332c841c3b76b/charts/bpndiscovery/templates/postgres-init.yaml#L38), for instance by introducing templates similar to [those](https://github.com/eclipse-tractusx/policy-hub/blob/main/charts/policy-hub/templates/_helpers.tpl#L72) (please define unique template names, like prefix every template `bpndiscovery`).

**Comment by tunacicek** [comment]
> Hi @tunacicek thanks for making available a chart version with configurable probes, so I was able to the get more logs and noticed that the discovery services aren't able to find their corresponding databases. This originates from [SPRING_DATASOURCE_URL](https://github.com/eclipse-tractusx/sldt-bpn-discovery/blob/26c372c86ae703df9bb43dcbd41332c841c3b76b/charts/bpndiscovery/templates/postgres-init.yaml#L38). Example: Locally I have chosen the release name `disc` which results in SPRING_DATASOURCE_URL: `jdbc:postgresql://disc-postgresql:5432/bpndiscovery` But the name of the database with the nameOverride (which is also necessary to set) is `disc-bpndiscovery-postgresql`. Once I change SPRING_DATASOURCE_URL to `jdbc:postgresql://disc--bpndiscovery-postgresql:5432/bpndiscovery` the service started up without issues. I recommend to change the way how you determine [SPRING_DATASOURCE_URL](https://github.com/eclipse-tractusx/sldt-bpn-discovery/blob/26c372c86ae703df9bb43dcbd41332c841c3b76b/charts/bpndiscovery/templates/postgres-init.yaml#L38), for instance by introducing templates similar to [those](https://github.com/eclipse-tractusx/policy-hub/blob/main/charts/policy-hub/templates/_helpers.tpl#L72) (please define unique template names, like prefix every template `bpndiscovery`).

Hi @evegufy , 
thanks for your feedback. I made the adjustments and the build runs successfully.

**Comment by tunacicek** [comment]
> Great that the helm install check runs now successfully. Could you please still add the option to disable the two components with a condition similar to those: [change in Chart.yaml](https://github.com/eclipse-tractusx/e2e-testing/blob/a489fd779f79b44308755fbe40410bb9442df1f4/charts/umbrella/Chart.yaml#L55) [change in values](https://github.com/eclipse-tractusx/e2e-testing/blob/a489fd779f79b44308755fbe40410bb9442df1f4/charts/umbrella/values.yaml#L153)

done.

**Review by almadigabor — APPROVED** [review]
Looks great, approved from my side.

**Review by evegufy — APPROVED** [review]

**Pull request merged**